lib/cusolver/dense_generic.jl

```diff
@@ -286,29 +286,15 @@ function Xgesvdp!(jobz::Char, econ::Int, A::StridedCuMatrix{T}) where {T <: Blas
     m, n = size(A)
     p = min(m, n)
     R = real(T)
-    econ ∈ (0, 1) || throw(ArgumentError("econ is incorrect. The values accepted are 0 and 1."))
-    U = if jobz == 'V' && econ == 1
-        CuMatrix{T}(undef, m, p)
-    elseif jobz == 'V' && econ == 0
-        CuMatrix{T}(undef, m, m)
-    elseif jobz == 'N'
-        CU_NULL
-    else
-        throw(ArgumentError("jobz is incorrect. The values accepted are 'V' and 'N'."))
-    end
+    econ == 0 || econ == 1 || throw(ArgumentError("econ is incorrect. The values accepted are 0 and 1."))
```

Comment:
Shouldn't `econ` be a simple booelan then?

Reply:
Yes, but I didn't wanted to break the API.
I was not smart when I did this high-level wrapper.
Maybe we can remove the restriction on the type and do `econ=Cint(econ)` inside the mathod.

Reply:
> `econ` should be a `int` if we follow the BLAS / LAPACK API

Our wrappers don't exactly follow the BLAS/LAPACK API, otherwise we'd just use the C wrappers, no?

Are you using these wrappers directly? I would have expected an additional layer of abstraction (e.g. `LinearAlgebra.svd!` with an `algorithm` argument), which is easier to promise stability for.

I'd switch to boolean and maybe add a deprecation:
```julia
if !(econ isa Bool)
  Base.depwarn(...)
  econ = (econ != 0)
end
```

Reply:
For the BLAS/LAPACK API, it depends on whether we are talking about the Fortran one (with all arguments) or the Julia one via `LinearAlgebra.LAPACK`.

The issue with the additional layer of abstraction is that `LinearAlgebra` lacks some high-level interfaces.  
This variant of the SVD is not supported by `LinearAlgebra.svd!`.

That's why I keep a few `LAPACK.*` methods here, we don't have another way to dispatch to them otherwise.  
I use these `LAPACK.*` methods in my packages because they provide the highest-level interface available for the same method on both CPU and GPU.

But I don't use `gesvdp!`, which is a quite recent routine in CUDA.